baremetal/metal3machine_manager.go

```diff
@@ -1345,12 +1344,15 @@ func (m *MachineManager) SetNodeProviderID(ctx context.Context, bmhID *string, p
 		m.Log.Info("unable to retrieve BMH name from Metal3Machine")
 		return &RequeueAfterError{RequeueAfter: requeueAfter}
 	}
-	providerIDLegacy := "metal3://unknown"
-	nodeLabel := fmt.Sprintf("%s=unknown", ProviderLabelPrefix)
-	if bmhID != nil {
-		providerIDLegacy = fmt.Sprintf("metal3://%s", *bmhID)
-		nodeLabel = fmt.Sprintf("%s=%s", ProviderLabelPrefix, *bmhID)
+	bmhID, err := m.getBmhIDFromM3Machine(ctx)
```

Comment:
even though the var had this naming before, let's fix it here and all other places (including logs) since it is not "ID" but "UID"

Reply:
I changed it in the places related to this PR, but there are some other places too where we are using bmhID instead of bmhUID. Should I change them too or create a separate PR for that?

Reply:
To keep the scope of this PR relevant, this should be good enough. Feel free to send a follow-up PR later.